webapp/components/test-file-results.js

```diff
@@ -270,8 +270,13 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
 
   resultsURL(testRun, path) {
     path = this.encodeTestPath(path);
-    // This is relying on the assumption that result files end with '-summary.json.gz'.
-    const resultsBase = testRun.results_url.slice(0, testRun.results_url.lastIndexOf('-summary.json.gz'));
+    // This is relying on the assumption that result
+    // files end with '-summary.json.gz' or '-summary_v2.json.gz'.
+    let resultsSuffix = '-summary.json.gz';
+    if (!testRun.results_url.contains('-summary')) {
```

Comment:
Won't the results_url always contain `-summary` for both the old format (`-summary.json.gz`) and the new format (`-summary_v2.json.gz`?

Also, what is the type for results_url?
javascript string has a method called `includes` but not `contains`
http://tpcg.io/_A9GPP0

Reply:
I'm realizing I did not hit the code to test this as I expected lol, Kyle also mentioned an issue with this above. I'll get it fixed and make sure it's 100% working.

Reply:
To help test, you may want to append to this [list](https://github.com/web-platform-tests/wpt.fyi/blob/207813b3ed18bae81068934caa478daffd782d36/webapp/components/test/util/helpers.js#L18) or create a TEST_RUNS_DATA_V2.

If you do the latter, you'll want to create a copy of this [file](https://github.com/web-platform-tests/wpt.fyi/blob/8bf23a6f68d18acab002aa6a613fc5660afb0a85/webapp/components/test/test-file-results-table.html) but load up TEST_RUNS_DATA_V2
https://github.com/web-platform-tests/wpt.fyi/blob/8bf23a6f68d18acab002aa6a613fc5660afb0a85/webapp/components/test/test-file-results-table.html#L29

Reply:
yes, I updated the `TEST_RUNS_DATA` you linked, and I edited some of the summary links to include a `_v2` suffix and left others with the old link. Since we're just validating that we can find the suffix as expected in the URLs, it should handle the test cases. Do you think this is sufficient? Or would it be best to update this further?

Reply:
We should add a test suite to the [test-file-results.html](https://github.com/web-platform-tests/wpt.fyi/blob/3f36cb0ac4f036e73049547f08242dda9d9e5637/webapp/components/test/test-file-results.html#L94) that exercises [resultsURL](https://github.com/web-platform-tests/wpt.fyi/blob/main/webapp/components/test-file-results.js#L271).

Something like:
```js
    suite('resultsURL', () => {
      test('v1 url', () => {
        expect(
          tfr.resultsURL(
            TEST_RUNS_DATA[1],
            "bar"
            )).to.equal("foo")
      });
      test('v2 url', () => {
        expect(
          tfr.resultsURL(
            TEST_RUNS_DATA[0],
            "bar"
            )).to.equal("foo")
      });
    });
```